src/input.cpp

```diff
@@ -377,18 +377,22 @@ int input_init() {
 
     input_common_init(&interrupt_handler);
 
-    const env_var_t term = env_get_string(L"TERM");
+    // Export variables used by curses into our own environment.
+    env_sync_env(L"TERM");
```

Comment:
This seems like the wrong way to do this. The var handling should be automatically calling setenv()/unsetenv() when exported vars are set or erased. I'm guessing that at the moment they exporting is only happening when an external program is run. Before merging this I need to understand why we're not keeping the two caches in sync in realtime.